### DISCUSSION TITLE
Error: cannot allocate vector of size 2.0 Gb

### DISCUSSION THREAD

**Discussion by shangyuan232** [statnet/latentnet#10] [Q&A] [answered]
Hi Pavel, I'm processing a bipartite network object converted from a 14000x9000 matrix. When I was running `net.fit <- ergmm(net~euclidean(d=2))`, it showed `Error: cannot allocate vector of size 2.0Gb`, then I increased the memory limit to 100,000, it showed  `Error: cannot allocate vector of size 3.9Gb` after 3 hours. I increased the memory limit again to 1,000,000, it showed `Error: cannot allocate vector of size 2.0Gb` again after 4 hours. 

What's the maximum network size that latentnet can process? Is there any way to get rid of the above error and make the calculation faster?

Thanks!

**Comment by krivit** [comment, marked as answer]
`latentnet`, unfortunately, uses sociomatrices to represent networks and dyadic covariates internally, and, furthermore, it uses the "expanded" bipartite representation. So, at some point it's going to allocate one or more massive matrices of integers and floating-point numbers.

I don't think I will be working on removing these limitations in the foreseeable future, but contributions would be welcome. Package https://cran.r-project.org/package=VBLPCM may be able to handle larger networks.
